community/detector/rce/CVE_2021_3129/src/main/java/com/google/tsunami/plugins/detectors/rce/cve20213129/Cve20213129VulnDetector.java

```diff
@@ -0,0 +1,147 @@
+      "{\"solution\":"
+          + " \"Facade\\\\Ignition\\\\Solutions\\\\MakeViewVariableOptionalSolution\",\"parameters\":"
+          + " {\"variableName\": \"cve20213129_tsunami\", \"viewFile\":"
+          + " \"phar://Tsunami_iDontExist\"}}";
+  private static final CharSequence DETECTION_STRING =
+      "file_get_contents(phar://Tsunami_iDontExist";
+
+  private final HttpClient httpClient;
+  private final Clock utcClock;
+
+  @Inject
+  Cve20213129VulnDetector(@UtcClock Clock utcClock, HttpClient httpClient) {
+    this.utcClock = checkNotNull(utcClock);
+    this.httpClient = checkNotNull(httpClient);
+  }
+
+  @Override
+  public DetectionReportList detect(
+      TargetInfo targetInfo, ImmutableList<NetworkService> matchedServices) {
+    logger.atInfo().log("Cve20213129VulnDetector starts detecting.");
+
+    return DetectionReportList.newBuilder()
+        .addAllDetectionReports(
+            matchedServices.stream()
+                .filter(NetworkServiceUtils::isWebService)
+                .filter(this::isServiceVulnerable)
+                .map(networkService -> buildDetectionReport(targetInfo, networkService))
+                .collect(toImmutableList()))
+        .build();
+  }
+
+  private boolean isServiceVulnerable(NetworkService networkService) {
+    String targetUri = NetworkServiceUtils.buildWebApplicationRootUrl(networkService) + QUERY_PATH;
+    try {
+      HttpResponse httpResponse =
+          httpClient.send(
+              HttpRequest.post(targetUri)
+                  .setHeaders(
+                      HttpHeaders.builder()
+                          .addHeader(CONTENT_TYPE, MediaType.JSON_UTF_8.toString())
+                          .addHeader(ACCEPT, MediaType.JSON_UTF_8.toString())
+                          .build())
+                  .setRequestBody(ByteString.copyFromUtf8(QUERY_PAYLOAD))
+                  .build(),
+              networkService);
+
+      return httpResponse.bodyString().orElseGet(() -> "").contains(DETECTION_STRING);
+
+    } catch (IOException e) {
+      logger.atWarning().withCause(e).log("Request to target %s failed", networkService);
+      return false;
+    }
+  }
+
+  private DetectionReport buildDetectionReport(
+      TargetInfo targetInfo, NetworkService vulnerableNetworkService) {
+    return DetectionReport.newBuilder()
+        .setTargetInfo(targetInfo)
+        .setNetworkService(vulnerableNetworkService)
+        .setDetectionTimestamp(Timestamps.fromMillis(Instant.now(utcClock).toEpochMilli()))
+        .setDetectionStatus(DetectionStatus.VULNERABILITY_VERIFIED)
+        .setVulnerability(
+            Vulnerability.newBuilder()
+                .setMainId(
+                    VulnerabilityId.newBuilder()
+                        .setPublisher("TSUNAMI_COMMUNITY")
+                        .setValue("CVE-2021-3129"))
+                .setSeverity(Severity.CRITICAL)
+                .setTitle("CVE-2021-3129: Laravel debug mode, unauthenticated RCE")
```

Comment:
Maybe update this to `CVE-2021-3129: Unauthenticated RCE in Laravel using Debug Mode`.